### DISCUSSION TITLE
How do I publish a standalone html that I can share with people?

### DISCUSSION THREAD

**Discussion by better-and-bad** [quarto-dev/quarto-cli#8867] [Q&A] [answered]
''''qmd
### Description

title: "Competitor Comparison"
format: 
  html:
    self-contained:true
editor: visual

I also used 
title: "Competitor Comparison"
format: 
  html:
    embed-resources:true
editor: visual
''''

When I render it to my viewer page, it looks great. I render it to my folder and open it locally and again it looks great. Not a stand along html doc, but as a webpage which is interesting. I'm really looking for an html doc that's not hosted as a webpage.

The real problem is that when I send it to someone, it loses the table of contents styling as well as other styling that made the html look presentable.

I'm really just trying to find a way to mimic how the html is originally rendered.

I tried opening a github pages section but I should just be able to export a simple html I'd imagine

````
base) Jacks-Air-3646:goodiebag jackconnors$ quarto check

[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.450
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.9 (Conda)
      Path: /Users/jackconnors/anaconda3/bin/python
      Jupyter: 5.2.0
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.0
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
      knitr: 1.43
      rmarkdown: 2.23

[✓] Checking Knitr engine render......OK

````

**Comment by mcanouil** [comment, marked as answer]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.

This being said `self-contained` is "deprecated" in favour of `embed-resources`.
This option create a "standalone" HTML file.

> it opens as just the raw code

It does not open by itself. You choose to open the HTML file and how you open it.
If you open it in an editor, it's to see the HTML code, so of course in this situation you see the code.
if you open it in a browser, then you see the rendered HTML.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by better-and-bad** [reply]
thank you, i hope my question is now presentable
